app/src/androidTest/java/com/epfl/drawyourpath/mainpage/fragments/ChallengeFragmentTest.kt

```diff
@@ -148,312 +191,159 @@ fun modifyPathGoalWithBlankDoesNothing() {
-
-    scenario.close()
-}
+        // change value of path
+        onView(withId(R.id.goals_view)).perform(
+            actionOnItemAtPosition<RecyclerView.ViewHolder>(
+                0,
+                replaceTextOnViewChild(newDistance.toString(), R.id.goal_display_edit_text),
+            ),
+            actionOnItemAtPosition<RecyclerView.ViewHolder>(
+                0,
+                pressImeActionButtonOnViewChild(R.id.goal_display_edit_text),
+            ),
+        )
 
-/**
- * test if the new time goal is correctly displayed when there is no today daily goal
- */
-@Test
-fun displayNewTimeGoalWhenNoTodayDailyGoal() {
-    val scenario = FragmentScenario.launchInContainer(
-        ChallengeFragment::class.java,
-        Bundle(),
-        R.style.Theme_Bootcamp,
-    )
-
-    val timeProgressGoal = context.resources.getString(R.string.progress_over_goal).format(0.0, mockUser.goals!!.activityTime!!)
-
-    waitUntilAllThreadAreDone()
-
-    // check that time is correct
-    onView(withId(R.id.goals_view))
-        .perform(scrollToPosition<RecyclerView.ViewHolder>(1))
-        .check(matches(hasDescendant(withText(timeProgressGoal))))
-        .check(matches(hasDescendant(withText(mockUser.goals!!.activityTime!!.toInt().toString()))))
-        .check(matches(hasDescendant(withText(R.string.minutes))))
-
-    scenario.close()
-}
+        val pathProgressGoal = context.resources.getString(R.string.progress_over_goal).format(0.0, newDistance.toDouble())
 
-/**
- * test if the nb of paths is correctly displayed when there is today daily goal
- */
-@Test
-fun displayNbPathsGoalOfTodayDailyGoal() {
-    val bundle = Bundle()
-    bundle.putBoolean(ChallengeFragment.TEST_EXTRA, true)
-
-    val scenario = FragmentScenario.launchInContainer(
-        ChallengeFragment::class.java,
-        bundle,
-        R.style.Theme_Bootcamp,
-    )
-
-    val pathProgressGoal =
-        context.resources.getString(R.string.progress_over_goal_path)
-            .format(dailyGoal.paths.toDouble(), dailyGoal.expectedPaths.toDouble())
-
-    waitUntilAllThreadAreDone()
-
-    // check that nb of paths is correct
-    onView(withId(R.id.goals_view))
-        .perform(scrollToPosition<RecyclerView.ViewHolder>(2))
-        .check(matches(hasDescendant(withText(pathProgressGoal))))
-        .check(matches(hasDescendant(withText(dailyGoal.expectedPaths.toString()))))
-        .check(matches(hasDescendant(withText(R.string.paths))))
-
-    scenario.close()
-}
+        waitUntilAllThreadAreDone()
+
+        // check that the value is correctly changed
+        onView(withId(R.id.goals_view)).check(matches(hasDescendant(withText(pathProgressGoal))))
+            .check(matches(hasDescendant(withText(newDistance.toString()))))
+            .check(matches(hasDescendant(withText(R.string.kilometers))))
+
+        scenario.close()
+    }
+
+    @Test
+    fun modifyTimeGoalDisplayNewTimeGoal() {

```

Comment:
Same here : This test is commented. Maybe add a TODO to explain the reason or delete it.